railties/lib/rails/application/bootstrap.rb

```diff
@@ -35,7 +35,7 @@ module Bootstrap
       # Initialize the logger early in the stack in case we need to log some deprecation.
       initializer :initialize_logger, group: :all do
         Rails.logger ||= config.logger || begin
-          logger = ActiveSupport::Logger.new(config.default_log_file)
+          logger = ActiveSupport::Logger.new(config.default_log_file, 1, config.log_size)
```

Comment:
```suggestion
          logger = ActiveSupport::Logger.new(config.default_log_file, 0, config.log_size)
```

`1` would keep the old log file around.

Reply:
However:

```ruby
Logger.new("/tmp/logs/test.log", 0, nil)
```
Won't log anything. So I think you need an `if config.log_size` conditional here.